internal/toproto6/deferred.go

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package toproto6
+
+import (
+	"github.com/hashicorp/terraform-plugin-go/tfprotov6"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+)
+
+func DataSourceDeferred(fw *datasource.Deferred) *tfprotov6.Deferred {
+	if fw == nil {
+		return nil
+	}
+	return &tfprotov6.Deferred{
+		Reason: tfprotov6.DeferredReason(fw.Reason),
+	}
+}
+
+func ResourceDeferred(fw *resource.Deferred) *tfprotov6.Deferred {
+	if fw == nil {
+		return nil
+	}
+	return &tfprotov6.Deferred{
+		Reason: tfprotov6.DeferredReason(fw.Reason),
+	}
+}
```

Comment:
I don't think these are hooked up to the relevant functions in:
- `internal/toproto6/importresourcestate.go`
- `internal/toproto6/planresourcechange.go`
- `internal/toproto6/readdatasource.go`
- `internal/toproto6/readresource.go`

Also, TIL that `golangci-lint`'s `unused` check doesn't apply for exported members in an `internal` package 😆 - https://github.com/dominikh/go-tools/issues/1406